src/app/panels/image-panel/image-panel.component.ts

```diff
@@ -1,13 +1,93 @@
-import { Component } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { EVTModelService } from 'src/app/services/evt-model.service';
 import { AppConfig } from '../../app.config';
 
 @Component({
   selector: 'evt-image-panel',
   templateUrl: './image-panel.component.html',
   styleUrls: ['./image-panel.component.scss'],
 })
-export class ImagePanelComponent {
+export class ImagePanelComponent implements OnInit  {
   manifest = AppConfig.evtSettings.files.manifestURL !== '' && !!AppConfig.evtSettings.files.manifestURL
     ? AppConfig.evtSettings.files.manifestURL
     : undefined;
+
+  get imageIndex() { return this._imageIndex; }
+
+  constructor(
+    public evtModelService: EVTModelService,
+    public http: HttpClient,
+    private cdref: ChangeDetectorRef,
+  ) {
+  }
+  public msDesc$ = this.evtModelService.msDesc$;
+  public secondaryContent = '';
+  private showSecondaryContent = false;
+  public msDescOpen = false;
+  public selectedPage;
+  public msDescID = '';
+  public value: number;
+  public images = [];
+  public currentImg: string;
+  // tslint:disable-next-line: variable-name
+  private _imageIndex: number;
+  public page = 0;
+
+  toggleImg(event){
+    this._imageIndex = this.images.indexOf(event);
+  }
+
+  eventHandler(event: number){
+    this.page = event;
+    this.currentImg = this.images[this.page].label;
+    this.cdref.detectChanges();
+  }
+
+  ngOnInit() {
+    // tslint:disable-next-line: no-any
+    this.http.get(`${this.manifest}`).subscribe((item: any) => {
+      item.sequences.forEach((value) => {
+        this.images.push(...value.canvases);
+      });
+      this._imageIndex = 0;
+      this.currentImg = this.images[this.page].label;
+      this.cdref.detectChanges();
+
+      return this.images;
+    });
+  }
+
+  getSecondaryContent(): string {
+    return this.secondaryContent;
+  }
+
+  isSecondaryContentOpened(): boolean {
+    return this.showSecondaryContent;
+  }
+
+  toggleSecondaryContent(newContent: string) {
+    if (this.secondaryContent !== newContent) {
+      this.showSecondaryContent = true;
+      this.secondaryContent = newContent;
+      this.msDescOpen = false;
+    }
+    else {
+      this.showSecondaryContent = false;
+      this.secondaryContent = '';
+    }
+  }
+
+  isMsDescOpen(event: boolean){
```

Comment:
This should be just `this.msDescOpen` why do you need the other lines?

Reply:
Weird to have a check on the internal state that depends on an event that is a boolean.
Check this function again, it is probably overcomplicated.